### PR TITLE
Batchnorm training mode support in a minimal build

### PR DESCRIPTION
**Pull request by baijumeswani**
Now that ONNX Runtime supports a minimal training build, the batchnorm in training mode is needed even in a minimal build.

This PR removes the preprocessor macro `BATCHNORM_INCLUDE_TRAINING_SUPPORT` and replaces it with `ENABLE_TRAINING_OPS`

### CODE REVIEW

**Review by edgchen1 — DISMISSED** [review]

**Review by edgchen1 — APPROVED** [review]

**Comment by baijumeswani** [comment]
Thanks for the review @edgchen1 :)

**Comment by BowenBao** [comment]
This is excluding batchnorm training mode support for standard onnxruntime releases, [regression found from ort-nightly==1.16.0.dev20230821001](https://hud.pytorch.org/pr/pytorch/pytorch/106984#16147600845). Can we revert that effect? There are some exported models for inference that runs just bn specifically in training mode.

**Comment by baijumeswani** [comment]
@edgchen1 @skottmckay 
Is it ok to completely remove the need for the preprocessor guard for training mode batchnorm?

@BowenBao under what conditions would you need the training batchnorm for inferencing?

**Comment by baijumeswani** [comment]
I created https://github.com/microsoft/onnxruntime/pull/17270

But I am not sure I understand why a training op would be needed in an inference build. @BowenBao could you use `onnxruntime_ENABLE_TRAINING_OPS` cmake flag in your test if needed or do we need the training bits in the inference build?

**Comment by BowenBao** [comment]
afaik some models are set to manipulate running mean and running variance in some ways. There is a separate issue https://github.com/pytorch/pytorch/issues/75252 discussing other ways of export but there is no traction. 

It would be nice if the previous behavior can be restored for 1.16 release. I hope training minimal build would still have this op after https://github.com/microsoft/onnxruntime/pull/17270 so the goal of this PR is not affected?
